### PR TITLE
[release/6.0][mono] add 400 band workloads

### PR DESCRIPTION
**Pull request by lewing**
this is to work around the broken sdk workload band install logic that has since been fixed, if it ships after August servicing we'll need to bump the runtime version.

### CODE REVIEW

**Comment by steveisok** [comment]
Kicked off official build for validation:

https://dev.azure.com/dnceng/internal/_build/results?buildId=1887920&view=results

File: eng/Versions.props [hunk 1]
```diff
@@ -25,7 +25,7 @@
   </PropertyGroup>
   <ItemGroup>
     <!-- The bands we want to produce workload manifests for -->
-    <WorkloadSdkBandVersions Include="6.0.100;6.0.200;6.0.300" />
+    <WorkloadSdkBandVersions Include="6.0.100;6.0.200;6.0.300;6.0.400" />
```
Comment (mmitche) [inline eng/Versions.props:28]: Since the 200 band SDK is no longer being built, is this a situation where we should be removing old workloads?

Comment (steveisok) [reply eng/Versions.props:28]: I'm not sure?  Would the customers who have a 200 SDK already installed not get the latest updates?

**Comment by steveisok** [comment]
Official build passed - the 400 manifests are being generated.

File: eng/Versions.props (hunk 1 above)
Comment (mmitche) [reply eng/Versions.props:28]: It's out of support, so would they be expected to get the latest updates?

**Comment by lewing** [comment]
@radical wbt is failing because there isn't a 400 band sdk yet

**Comment by radical** [comment]
hm I can add a fix for this.

**Comment by radical** [comment]
`Microsoft.NET.Workload.Emscripten.Manifest-6.0.400` nuget is missing. I don't see a flow PR for this either.

`
  /tmp/install-workload/o11ldwim.qvg/restore/Restore.csproj : error NU1101: Unable to find package Microsoft.NET.Workload.Emscripten.Manifest-6.0.400. No packages exist with this id in source(s): artifacts-nonshipping-packages, artifacts-shipping-packages, darc-pub-dotnet-emsdk-52e9452, dotnet-eng, dotnet-public, dotnet-tools, dotnet6, dotnet6-transport, richnav
`

**Comment by radical** [comment]
`workload install` is installing `6.0.2-mauipre.1.22102.15` version for the packs, which it gets from the mono.toolchain manifest for `6.0.300` .
```
  ** workload install **

  Running: /workspaces/runtime/artifacts/bin/dotnet-workload/dotnet workload install --skip-manifest-update --no-cache --configfile "/tmp/elz3pk5g.cfm" wasm-tools
  Using working directory: /tmp/
  Installing pack Microsoft.NET.Runtime.WebAssembly.Sdk version 6.0.2-mauipre.1.22102.15...
  Skip NuGet package signing validation. NuGet signing validation is not available on Linux or macOS https://aka.ms/workloadskippackagevalidation .
  Writing workload pack installation record for Microsoft.NET.Runtime.WebAssembly.Sdk version 6.0.2-mauipre.1.22102.15...
  Installing pack Microsoft.NETCore.App.Runtime.Mono.browser-wasm version 6.0.2-mauipre.1.22102.15...
....
                     Garbage collecting for SDK feature band(s) 6.0.300... (TaskId:444)
                     Successfully installed workload(s) wasm-tools. (TaskId:444)
```

**Comment by radical** [comment]
`dotnet workload install` is installing for sdk band `6.0.300`. Is this controlled in the SDK, and isn't there yet?

**Comment by radical** [comment]
AFAICS, this `dotnet workload install` is falling back to `6.0.300` because it can't find the manifest for Emscripten/6.0.400 . If I install with `--sdk-version 6.0.400`:

```
/workspaces/runtime/src/libraries/workloads-testing.targets(113,5):
error : workload install failed: Unhandled exception: Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestCompositionException: Did not find workload manifes
t dependency 'Microsoft.NET.Workload.Emscripten' required by manifest 'microsoft.net.workload.mono.toolchain' [/workspaces/runtime/artifacts/bin/dotnet-workload/sdk-manifests/6.0.400/microsoft.net.workload.mono.toolchain/WorkloadM
anifest.json] [/workspaces/runtime/src/tests/BuildWasmApps/Wasm.Build.Tests/Wasm.Build.Tests.csproj]
/workspaces/runtime/src/libraries/workloads-testing.targets(113,5): error :    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.ComposeWorkloadManifests() [/workspaces/runtime/src/tests/BuildWasmApps/Wasm.Build.Tests/W
asm.Build.Tests.csproj]
```

**Comment by lewing** [comment]
flow is turned off for emsdk right now, I assume that was for release management

File: eng/Versions.props (hunk 1 above)
Comment (lewing) [reply eng/Versions.props:28]: there is some history here and I'm not sure I have it right but the way I understood things for workloads is that 200 is intended to see 100 band updates as a fallback but due to some sdk bugs we've had to keep producing multiple bands as workarounds.  Given that it seemed like we'd be matching the intended design if we if we make sure 200 still sees any changes that apply to 100.

**Comment by lewing** [comment]
https://github.com/dotnet/runtime/pull/72616

**Comment by lewing** [comment]
failures are unrelated, do we plan to take this?

**Review by steveisok — APPROVED** [review]

**Pull request merged**